ARExperiment/ARExperiment/LightsAnimationsViewController.swift

```diff
@@ -87,53 +62,71 @@ class LightsAnimationsViewController: UIViewController, ARExperimentSessionHandl
 extension LightsAnimationsViewController: ARSCNViewDelegate {
     
     func renderer(_ renderer: SCNSceneRenderer, didAdd node: SCNNode, for anchor: ARAnchor) {
-        if !anchor.isKind(of: ARPlaneAnchor.self) {
-            DispatchQueue.main.async { [weak self] in
-                guard let strongSelf = self else { return }
-                guard let model = strongSelf.earthNode,
-                    let secondModel = strongSelf.moonNode else {
-                        print("We have no model to render")
-                        return
+        guard !anchor.isKind(of: ARPlaneAnchor.self) else {
+            return
+        }
+        DispatchQueue.main.async { [weak self] in
+            guard let strongSelf = self else { return }
+
+            for assetNode in strongSelf.arAsset.nodesOfType(.model) {
```

Comment:
I'd recommend extracting this as a method. Long async blocks become really hard to read. I _always_ split the threading and actual work:

```swift
DispatchQueue.main.async { [weak self] in
    guard let strongSelf = self else { return }
    strongSelf.doActualWork()
}
```

Reply:
Looking at why I still needed this dispatch and how to clean up the code I realised in some cases I was still getting the error mentioned on the PR. That is because what Apple does not like is any `SCN` being accessed while on a rendered method. That means we need to create the `SCNNodes` before we get to render anything on the screen. 

We refactored `ARViewModel` to basically be in charge of creating an array of SCNNodes that each controller will need when the tracking starts. This turns up really well because now no controller needs repeated code of creating nodes and setting them properly since the only thing render methods need to do now is grabbing them from an array and adding them to the scene. 

You can see how we create the nodes on view model for every possible type of node here -> https://github.com/novoda/ios-demos/pull/36/commits/dc1c38741784a07ff8756df053725690f6ddbab2 and how is that implemented and helped clean up the controllers code here -> https://github.com/novoda/ios-demos/pull/36/commits/8beb12a9a8bac12d6c4f7aa3a603ccf079d5f83a 

For the light nodes we need the light estimation before we can create the node, and that means that we need to only create those when session is at normal tracking. So we added a method to the session handler to let the controllers know when that is happening and create the nodes then -> https://github.com/novoda/ios-demos/pull/36/commits/56f023501f65ae0fc85c8a03fa23e429698c2673 

Also I checked and it seems on the rendered method we are already on main so the dispatch was not needed. What do you think @amlcurran ?